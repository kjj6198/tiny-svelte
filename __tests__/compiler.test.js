const parse = require("../index");
const Component = require("../compile/Component");

describe("Component", () => {
  it("should compile", () => {
    const ast = parse(`<script>console.log()</script><p>html</p>`);
    const component = new Component(ast);
    expect(component.fragment.children[0].type).toBe("Element");
    expect(component.fragment.children[0].children[0].type).toBe("Text");
  });

  it("should compile js", () => {
    const ast = parse(`<script>console.log()</script><p>html</p>`);
    const component = new Component(ast);
    expect(component.generate()).toMatchSnapshot("");
  });

  it("should compile js (nested)", () => {
    const ast = parse(
      `<script>console.log(); let a = 1</script><p>html <span>123</span></p>`
    );
    const component = new Component(ast);
    expect(component.generate()).toMatchSnapshot();
  });

  it("should compile js (mustachetag)", () => {
    const ast = parse(
      `<script>console.log(); let a = 1</script><p>html <span>123 {a}</span></p>`
    );
    const component = new Component(ast);
    expect(component.generate()).toMatchSnapshot();
  });

  it("should compile js (attr)", () => {
    const ast = parse(
      `<script>console.log(); let a = 1</script><p role="status">html</p>`
    );
    const component = new Component(ast);
    expect(component.generate()).toMatchSnapshot();
  });
});
