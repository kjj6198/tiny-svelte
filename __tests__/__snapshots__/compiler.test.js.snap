// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should compile js (attr) 1`] = `
"let a = 1;
function instance() {
	console.log();
}

function create_fragment() {
	let p1;
	let t1;

	return {
		create: () => {
			p1 = document.createElement(\\"p\\");
			t1 = document.createTextNode(\\"html\\");
		},
		mount: () => {
			p1.setAttribute(role, \\"status\\");
			append(p1, fragment);
			append(t1, p1);
		}
	};
}"
`;

exports[`Component should compile js (mustachetag) 1`] = `
"let a = 1;
function instance() {
	console.log();
}

function create_fragment() {
	let p1;
	let t1;
	let span1;
	let t2;

	return {
		create: () => {
			p1 = document.createElement(\\"p\\");
			t1 = document.createTextNode(\\"html \\");
			span1 = document.createElement(\\"span\\");
			t2 = document.createTextNode(\\"123 \\");
		},
		mount: () => {
			append(p1, fragment);
			append(t1, p1);
			append(span1, p1);
			append(t2, span1);
			append(a, span1);
		}
	};
}"
`;

exports[`Component should compile js (nested) 1`] = `
"let a = 1;
function instance() {
	console.log();
}

function create_fragment() {
	let p1;
	let t1;
	let span1;
	let t2;

	return {
		create: () => {
			p1 = document.createElement(\\"p\\");
			t1 = document.createTextNode(\\"html \\");
			span1 = document.createElement(\\"span\\");
			t2 = document.createTextNode(\\"123\\");
		},
		mount: () => {
			append(p1, fragment);
			append(t1, p1);
			append(span1, p1);
			append(t2, span1);
		}
	};
}"
`;

exports[`Component should compile js 1`] = `
"
function instance() {
	console.log();
}

function create_fragment() {
	let p1;
	let t1;

	return {
		create: () => {
			p1 = document.createElement(\\"p\\");
			t1 = document.createTextNode(\\"html\\");
		},
		mount: () => {
			append(p1, fragment);
			append(t1, p1);
		}
	};
}"
`;
